### ISSUE TITLE
Feature - Removed from Watchlist

### ISSUE THREAD

**Issue by animosity22** [Radarr/Radarr#1362]
My general flow is to add items to my trakt watchlist as time goes on and once something is grabbed and downloaded, it was normally removed from the watchlist that it was on.

I tend to remove items I already have and move them away to cloud storage or something to help save on local space. By having to manually remove it from the list, it breaks some of the nice automation flow that normally goes on.

**Comment by geogolem** [comment]
I think the feature you are requesting already exists....

I basically use trakt to manage everything.
----------------------------
Trakt Watchlist: What I want to watch
Trakt Watched List: What I have watched (history etc.)
Trakt Collected List: What I have downloaded and currently have available (i.e. what is currently available in my Plex Library)

Plex can add/remove things from my Trakt Collected List as items are added/removed from my Plex Library. 

Plex also pulls my watched list history from trakt and will add items to my trakt history (watched list) as I watch them. Plex does not touch my Trakt Watchlist directly (though as items are marked as Watched they will be added to my Trakt Watched List and trakt will remove from the WatchList).

Plex will never remove anything from my Trakt Watched List (though it will add to it). 

Radarr auto imports my trakt watchlist.... and I use the "Clean Library - log Only" option.. so if an item is in radarr (downloaded or not) that is not in my trakt watch list it will output a message in the log.

The idea is that I can totally manage everything from trakt. If I want to watch a movie, I add it to my trakt watchlist. Radarr will add that movie, download it and add it to the Plex library. Plex will then add it to my Trakt Collected List.

I eventually might watch the movie on my TV via Plex. When i finish watching it --> Plex will add it to my Trakt Watched List and it will be removed from my watchlist...

Radarr will spit out a message indicating that this movie is in Radarr but not in my trakt watchlist. If I choose I can delete the movie from radarr (with or without files) when I delete the files, Plex will remove the item from my trakt collection etc.

I only use the logOnly option right now - since I dont want radarr to accidentally remove movies because of a bug or trakt list sync issue.... but theoretically I could have radarr automatically remove the movies  -- always keeping radarr's movie list as a copy of my trakt watchlist.

---------------------

I also have a "netflix" feature in my personal build... Any movies that are in my trakt watchlit that are currently available on Netflix will still be added to radarr but will be marked as "Unmonitored" so they wont be downloaded... --> Since i have a netflix subscription I find this useful --> for now its a personal feature.

-------------------

**Comment by geogolem** [comment]
I think the clean library feature is what you are requesting.. but Im not sure if I am misunderstanding your request. Please let me know if you need help with this feature or clarify what exactly you would like to accomplish.

**Comment by geogolem** [comment]
--> Perhaps I did misunderstand what you wanted....  Is it that you want the item to be removed from the "Trakt Watchlist" when it is downloaded?

though I can see someone using the watchlist in this way --> the trakt watchlist is actually supposed to (if i understand correctly) reflect what movies you would like to watch.... including movies that you might have downloaded but have yet to watch. Trakt has a built in feature to automatically remove movies from the watchlist when they are added to the Trakt Watched List...

What you seem to want is to have movies removed from the Trakt Watchlist when they are added to the Trakt Collection....  

I think this is an unorthodox usage of the Trakt Watchlist... BUT I do understand your idea of moving movies away to cloud storage or something... and thus not wanting them in your radarr library... 

Radarr does have the ability to specify that you want to ignore items from future list imports whenever you delete them... 

It also has the ability to automatically unmonitor movies whenever the files are discovered to be missing  (i.e. they were deleted)

![image](https://cloud.githubusercontent.com/assets/23205277/24939482/a3e949e8-1f0b-11e7-9fd5-d5f462bb7248.png)

**Comment by animosity22** [comment]
Sorry explained so poorly.

Nothing in Radarr directly, but it pulls from a trakt watchlist and there was an option to remove it from the watchlist once downloaded.

So workflow would be movie added to trakt watchlist called Blah.
Radarr grabs movie from watchlist and adds Blah to get downloaded
Once downloaded, Blah is removed from Trakt watchlist so it cleans up the watchlist.

**Comment by animosity22** [comment]
Specifically in Couchpotato, there is a "Remove from Watchlist" option once it gets downloaded.

**Comment by geogolem** [comment]
I see --> I understand what you want.. So you are not using the Trakt watchlist to track which movies you want to watch but rather which movies you want to download?

**Comment by animosity22** [comment]
I do that as well. The watchlist populates movies I want to download. The 'watch' is a different flag.

Once "Marked as Done" in Couchpotato, it removes from the Watchlist only.

In this shot, you can see I "collected it' (it's a the wrong resolution so still in my list). Once marked done, it removes from the Watchlist.

http://animosity022.imgur.com/all/

**Comment by geogolem** [comment]
I can see how people might want to use a feature like this but its not really how the Trakt WatchList is intended to be used IMO (but then again Trakt isnt really intended to be used as input to something like RADARR.. is it lol...)..

I see someone on trakt tried to request a similar feature: http://support.trakt.tv/forums/188762-general/suggestions/5200378-remove-collected-movies-from-watchlist

Im not trying to be difficult by having this discussion Im just trying to figure out the best way to think about/implement what you want... Radarr doesn't always want to do everything couchpotato did or do it in the same way...

i.e. Radarr supports other lists other than just the watchlist...

will post another comment in second. need to think about something..

**Comment by animosity22** [comment]
I used it as it just was there in Couchpotato.

It's like a backup of the current "Missing/Wanted" section so if you had to redo, it's instant.

**Comment by geogolem** [comment]
yes - i personally use it the opposite way... my radarr is like a backup of my trakt watchlist.. so if i had to reinstall radarr etc. everything would be back in radarr as it was before.. i.e. do a list import and then scan the disk for files etc.

I understand what couchPotato did... and I understand that you effectively want the same behavior out of radarr and I dont think its a bad idea --> we just have to discuss to determine the best way to do it in radarr.. having a simple option "remove from trakt watchlist upon download" does not really translate to radarr since radarr allows importing of other lists from trakt.. not just the watchlist.. and radarr also has filtering etc.

**Comment by geogolem** [comment]
For example, in Radarr I am thinking the best way to do this would be to have a new option in the Edit List dialog:

![image](https://cloud.githubusercontent.com/assets/23205277/24939887/6263da4e-1f0e-11e7-8eb0-d2b7f205fce6.png)


Remove item from list on successful Download

**Comment by animosity22** [comment]
Yep. That would meet the use-case I was doing without a problem.

I don't offload to my ACD drive until it's actually done so that works perfect.

**Comment by geogolem** [comment]
So basically upon every successful download in Radarr --> any lists with "Remove item from list on successful Download" set to YES.... will have the corresponding movie removed (if it exists).. If you have multiple lists and the movie exists on multiple lists... it will be removed from all the lists that has this turned on.. Does that make sense?

**Comment by geogolem** [comment]
If thats the case I think its definitely a useful feature and feasible.

**Comment by animosity22** [comment]
Yep. You got it. And definitely no rush as I just wanted to get the idea out there :)

**Comment by geogolem** [comment]
So I did some investigation into this --> and with some more information/experience I think perhaps an implementation like CouchPotato is more appropriate. You can only remove items from the watchlist for which Radarr is currently authenticated...  

so this should correspond to the authentication --> and not the list itself.

For custom trakt lists the method previously described should work but for the watchlist.. it makes more sense to attach the option with the "authentication"

**Comment by geogolem** [comment]
perhaps also, you might want to have the option --> remove from watchlist when "cutOff Met" or something like that, also... anyway, if we implement it in one spot (i.e. on successful Import) it should be quite easy to make it happen on another event.. like cutoff met or something. etc.

**Comment by animosity22** [comment]
Yes, that does work well too. I'm trying to adjust to the Radarr terminology for events as "marked as done" was more Couchpotato, but Cutoff is definitely good.

I only use the 1 trakt watchlist for my purposes. My other trakt use is a backup of my library watched status so that's my main goal out of Trakt.

I tend to browse the trending stuff and add items to my watchlist so pick up new movies.

**Comment by raspdealer** [comment]
Please not that trakt can do the job, but only removed fron watchlist when watched, not when collected. You should request this feature in tralt in my opinion.

**Comment by animosity22** [comment]
@raspdealer  - that doesn't seem correct as Couchpotato removes when collected. You might not watch something for weeks/months but have it in your library.

**Comment by jkaberg** [comment]
I agreed with @animosity22, and in addition to what the PR already does I'd love to see Radarr also marking the movie as collected in the same watchlist (untag "watchlist", tag "collected"). Obviously made optional, as not all would like that.

**Comment by ideaprison** [comment]
Another user here that desperately needs this. Just like how couchpotato does this. I can't use Radarr till this has been added, else it's just too much manual work. Other than that feature missing Radarr is great!

**Comment by jkaberg** [comment]
Perhaps @galli-leo could review this one? 😄

**Comment by christiga** [comment]
I defintely would like radarr to send info to Trakt as movie has been "collected". Thanks

**Comment by jokerigno** [comment]
I would love this feature!

**Comment by HitsvilleUK** [comment]
Hs there been any more thoughts on this?

**Comment by geogolem** [comment]
I havent been working on radarr for quite a while... but noticed all these messages. I understand that people want to be able to add items to the trakt "collected" list when they have been downloaded in radarr... I think its likely quite feasible to implement such a feature in radarr but Im not sure when I could look more into it (or someone else)... 

I personally do require something similar in my set-up /workflow; however, I use PLEX to manage my library.

I maintain a trakt watchlist which radarr uses to determine which movies I want to download. Once radarr downloads a movie PLEX notices and PLEX updates my Trakt "Collected" list. When I watch it, Plex also updates my trakt "Watched List" and when the item is added to my trakt "Watched List" trakt automatically removes it from my trakt "watchlist".

if I delete the movie from radarr and from disk -- Plex notices and removes it from my trakt "Collected" list.

Im not sure if you guys are using PLEX but here is a link to the plugin that does the trick for me: https://github.com/trakt/Plex-Trakt-Scrobbler

**Comment by HitsvilleUK** [comment]
Hey thanks for the reply geogolem. 
Yes I do use Plex and the Plex-Trakt-Scrobbler.
But as I'm about to type the reasons i "thought" I really needed this feature its become muddied. 
I'm a data hoarder and deal mainly with Remuxes be they 1080p and more recently 4K.

I guess the issue stems from wanting to grab both the 1080p Remux (for the people I share my server with as well as myself) and the 4K Remux (for solely personal use) when it becomes available. 
I guess my initial need was because Radarr cannot KEEP multiple qualities but I guess the fact I just realized I can create a 1080p AND a 4K watchlist in Trakt negates the issue.

I have a local Windows Plex install and a dedicated Linux server with Gdrive mounted. 
In Linux I have managed to get 2 instances of Radarr running and guess I can have each instance looking at a particular list and each using there own "label" in the torrent client. Then they both send the respective quality to the correct directory on Gdrive.
Windows as far as I'm aware I can only have one instance running, so may see if I can get around this with a VM. 
Sorry for bumping up the feature request as I was probably going down the wrong track with my thoughts

**Comment by geogolem** [comment]
Though these issues are different a distinct - implementation details are likely highly related: https://github.com/Radarr/Radarr/issues/3014

**Comment by geogolem** [comment]
Although slightly different the request is similar: https://github.com/Radarr/Radarr/issues/3139

**Comment by TeddOravec** [comment]
Hi there!
I know this is a hella old issue and was just wondering if there has been any movement on this recently or if I should seek alternative options. I would also like to be able to have Radarr remove the item from the Trakt list when it finds a suitable download, so I can use my custom trakt list as a remote queuer of sorts. Thanks for the good work!

**Comment by GodsVictory** [comment]
Any updates on this?

**Comment by NGDM** [comment]
Also pushing for this change. 
I have a couple personal lists on Trakt which are mapped to different Radarr folders.
When I import or watch one of the movies I would love it to remove the movie from the personal list it originated from. 

Now I indeed need to use the Import Exclusions but that's just not practical if I ever want to rewatch/redownload a movie which I excluded before…
This feature would override this issue, so big **+1**

**Comment by geogolem** [comment]
Recently there was a TraktConnection added to Radarr which allows one to update their Trakt Collection when movies are downloaded or deleted from Radarr. There are still some aspects which are a work in progress. 

You can find more information about this issue and its related PR's here: https://github.com/Radarr/Radarr/issues/4521

This doesn't solve the original feature request in this ISSUE; however it does address secondary requests mentioned throughout..

perhaps a generalization could be added to the new TraktConnection feature, that allows the user to specify the list type etc. so that the connectron type is not just permitted for the Collection Syncing but could also be used to for example, remove items from the TraktWatchList upon successful download. 

it would just require the user specify more information in the Connection.

User could create one TraktConnection for Collection, which add's OnImport, OnUpgrade and removes OnDelete

User could create one TraktConnection  for WatchList which removes OnImport...

Basically the Trakt Connection would have to make it user specifiable which List (WatchList, Collection etc.), and what to do  (add/remove etc.) on which events.

**Comment by rg9400** [comment]
The ability to delete an item from a list can be solved via a custom post-processing script until it is added natively. Find such an example right here (radarr_delete_list.py)

https://github.com/rg9400/Trakt-Arr-Integration

**Comment by bakerboy448** [comment]
better covered under https://github.com/Radarr/Radarr/issues/4521

**Comment by GodsVictory** [comment]
> better covered under #4521

I'd like to argue this is similar, although distinctly different than #4521.

The request is to remove items from a trakt list, while #4521 is requesting the ability to add downloaded movies to a collection.
